### ISSUE TITLE
Improved color management

### ISSUE THREAD

**Issue by jcwinkler** [textext/textext#21]
Currently the colors defined in the LaTeX code are used for displaying the node in Inkscape.

Some users like to assign the colors in Inkscape. When the node is recompiled this color assignment is lost. 

For the user it would be nice to have the possibility to decide if the color defined in Inkscape is preserved after recompiling or if the colors from the TeX code is used.

**Comment by sizmailov** [comment]
Is it supposed to handle one color per textext object or its should handle colored elements too? 

Say I want to highlight sine argument `x`. What should I get after replacement `x` by `x+1`? What should be colored `x` or `x+1`?
![image](https://user-images.githubusercontent.com/2975991/39799696-d5f4652a-536d-11e8-9638-f602bcf9f28e.png)

I wonder how it might be implemented in general and reliable way? To do that you need to know one-to-one correspondence between children of old and new node, but this information is not available to textext.

**Comment by jcwinkler** [comment]
What I am more worried about is the following behavior in pdf2svg nodes: After assigning the fill color to a node one may obtain the following:
![grafik](https://user-images.githubusercontent.com/11866252/39801594-02b2525e-536c-11e8-8f57-43a082da6555.png)

To get the root completely colored one needs to also set the stroke color as in pstoedit), but this leads to:

![grafik](https://user-images.githubusercontent.com/11866252/39801641-246ca494-536c-11e8-84b3-aa6ba1ff3b93.png)

Reason is the path desribing the bar of the root:
```
        <path
           id="path29"
           transform="matrix(1,0,0,-1,182.358,126.263)"
           d="M 0.001375 0.00128125 L 30.657625 0.00128125 "
           style="fill:none;stroke-width:0.398;stroke-linecap:butt;stroke-linejoin:miter;
                      stroke:rgb(0%,0%,0%);stroke-opacity:1;stroke-miterlimit:10;" />
```

which differs from the other elements which are of type

```
        <g
           id="g27"
           style="fill:rgb(0%,0%,0%);fill-opacity:1;">
          <path
             id="path25"
             transform="translate(174.056,126.462)"
             d="M 3.875 ... "
             style="stroke:none;" />
        </g>
```

Adding `stroke-width:0;` to each path-element which has `style="stroke:none;` resolves the problem. But I do not know if this is valid svg code.

**Comment by jcwinkler** [comment]
(It would be easy to do this in the `svg_to_group` method of `Pdf2SvgPlotSvg` class.)

**Comment by sizmailov** [comment]
`stroke-width:0;` is a valid statement according to

https://www.w3.org/TR/fill-stroke-3/#stroke-width

> This property specifies the width of the strokes on the outline. A zero value causes no stroke to be painted for that layer. Negative values are invalid.

**Comment by jcwinkler** [comment]
OK. I was just confused about the combination `stroke:none` and `stroke-width:0` wrongly assuming `stroke:none` means no stroke is painted. But `none` refers to the color, of course. Forgot that.

So I will implement it in the `svg_to_group` method.

**Comment by jcwinkler** [comment]
GUI for creating new nodes:
![grafik](https://user-images.githubusercontent.com/11866252/39817314-1b1b4d46-539e-11e8-9c47-1b69300eb814.png)

GUI for editing nodes:
![grafik](https://user-images.githubusercontent.com/11866252/39817375-472dd76e-539e-11e8-80c6-b7f4e8c98be1.png)

Alternative: Let the color checkbox always be sensitive. When creating new nodes this should mean that the output is black even if color commands are in the TeX code. Maybe nice when copying colored TeX code from somewhere and don't want to have the color in the document... ??

**Comment by jcwinkler** [comment]
> Adding stroke-width:0; to each path-element which has style="stroke:none;" resolves the problem.

Introduced via commit 348305c

**Comment by jcwinkler** [comment]
> I do not thinks that overloading one control with several meanings is a good idea.
> 
> ... so I would stay with your current implementation.

:+1:

**Comment by sizmailov** [comment]
"Keep Inkscape color" should be enabled and checked if and only if user changed current node color in inkscape. Otherwise this option is meaningless.

**Comment by jcwinkler** [comment]
Alternative idea, without the need of any additional GUI element:

1. When a new TexText object is created the colorization coming from the TeX code/ SVG converter is used. This is obvious, of course.
2. When an existing TexText object is re-edited the following decision is done automatically:
   - If the TeX code contains color commands, any colorization done in Inkscape is ignored when placing the edited node in the document
   - If the TeX code does not contain color commands, the style from the existing node is applied (as in previous versions of TexText, `copy_style` is called), so colorization done in Inkscape is kept.

I think this is quite an natural approach because the intention of the user to have  TeX colors is already expressed by his color commands and must not be additionally expressed by clicking or pressing anything.

Color commands in the TeX codes could be identified by two approaches:

1. Search the TeX code for color commands like \usepackage{color}, \textcolor, etc. (-> not 100% safe, I think)
2. Search the SVG code coming from the SVG converter for colors \neq #000000.

What do you think?

**Comment by sizmailov** [comment]
Good idea. I find it very convenient. 
I would not rely on parsing user input either. Inspecting resulting svg is straightforward and reliable.

**Comment by leberwurstsaft** [comment]
Definitely only look in the resulting SVG. One can never be sure to catch all the ways a LaTeX-document adds color.
I wonder why pdf2svg and pstoedit generate such different SVG results, though.

**Comment by jcwinkler** [comment]
Fixed in release 0.8.
